pandas/_libs/join.pyx

```diff
@@ -410,6 +411,7 @@ def left_join_indexer(ndarray[join_t] left, ndarray[join_t] right):
 left_join_indexer_float64 = left_join_indexer["float64_t"]
 left_join_indexer_float32 = left_join_indexer["float32_t"]
 left_join_indexer_object = left_join_indexer["object"]
+left_join_indexer_int8 = left_join_indexer["int8_t"]
```

Comment:
Do we need int16 too? That can happen with

```python
In [21]: c = pd.Categorical(np.arange(np.iinfo('int16').max - 1))

In [22]: c.codes.dtype
Out[22]: dtype('int16')
```

Reply:
I changed also for dtype int16 in the new commit